### ISSUE TITLE
Preserve the original request URL in "onUnhandledRequest" warnings

### ISSUE THREAD

**Issue by kettanaito** [kettanaito/msw#2125]
### Prerequisites

- [X] I confirm my issue is not in the [opened issues](https://github.com/mswjs/msw/issues)
- [X] I confirm the [Frequently Asked Questions](https://mswjs.io/docs/faq) didn't contain the answer to my issue

### Environment check

- [X] I'm using the [latest](https://github.com/mswjs/msw/releases/latest) `msw` version
- [X] I'm using Node.js version 18 or higher

### Node.js version

v20.11.0

### Reproduction repository

Always specify repros. Don't follow my lead.

### Reproduction steps

1. Set up MSW in the browser.
2. Configure the `onUnhandledRequest` callback:
```ts
await worker.start({
  onUnhandledRequest(request, print) {
    console.log('URL:', request.url)
    if (/\.(css|js|json|png|jpg|gif)$/.test(request.url)) {
      return
    }

    print.warning()
  },
})
```
3. Make a request like `/fo/bar/?t=123`

### Current behavior

MSW prints a clean URL, making it hard to understand why my custom `onUnhandledRequest` logic is not matching this request:

![Screenshot 2024-04-05 at 15 45 08](https://github.com/mswjs/msw/assets/14984911/de3bcb24-d824-494e-8cfb-e7d611eabcbe)

### Expected behavior

MSW preserves the unhandled request URL as-is, which includes keeping its query parameters and fragments.

**Comment by kettanaito** [comment]
## How to fix this

The fix would be right here: 

https://github.com/mswjs/msw/blob/e83a6699bc40a4fc2a593a93346158b8b8b61649/src/core/utils/request/onUnhandledRequest.ts#L25

The reason we use `toPublicUrl()` here is to omit the document's origin and print a relative URL. There are two ways to fix this:

1. Always print an absolute URL. I.e. remove `toPublicUrl` and rely on `request.url` as-is.
2. Still get a potentially relative URL but copy the `search` and `fragment` from the original `request.url` (we construct a URL instance out of it anyway).

**Comment by nagadevkrishna** [comment]
Hi, I made a change and raised the PR: https://github.com/mswjs/msw/pull/2128

**Comment by kettanaito** [comment]
## Released: v2.2.14 🎉

This has been released in v2.2.14!

- 📄 [**Release notes**](https://github.com/mswjs/msw/releases/tag/v2.2.14)
- 📦 [npm package](https://www.npmjs.com/package/msw/v/2.2.14)

Make sure to always update to the latest version (`npm i msw@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
